auto_nag/auto_mock.py

```diff
@@ -0,0 +1,125 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This file was originally copied from https://github.com/mozilla/libmozdata/blob/v0.1.82/tests/auto_mock.py
+
+import gzip
+import hashlib
+import os
+import pickle
+import re
+import unittest
+from typing import List
+from urllib.error import HTTPError
+from urllib.parse import parse_qsl, urlparse
+from urllib.request import Request, urlopen
+
+import responses
+
+from auto_nag import logger
+
+MOCKS_DIR = os.path.join(os.path.dirname(__file__), "tests/mocks")
+
+
+class MockTestCase(unittest.TestCase):
+    """
+    Mock responses from any webserver (through requests)
+    Register local responses when none are found
+    """
+
+    mock_urls: List[str] = []
+
+    def setUp(self):
+
+        # Setup mock callbacks
+        for mock_url in self.mock_urls:
+            url_re = re.compile(rf"^{mock_url}")
+            responses.add_callback(
+                responses.GET,
+                url_re,
+                callback=self._request_callback,
+                content_type="application/json",
+            )
+
+    def _request_callback(self, request):
+        logger.debug("Mock request %s %s", request.method, request.url)
+        path = self._build_path(request.method, request.url)
+
+        if os.path.exists(path):
+            # Load local file
+            logger.info("Using mock file %s", path)
+            with gzip.open(path, "rb") as file:
```

Comment:
Maybe we should not gzip, to have only readable files in the repo and normal diffs.
There aren't too many mock files, so we are not gaining much in compressing them.

Reply:
Done in 954ec05e0ba3cc94561822fcec883cc74ab3c105